chia/data_layer/data_layer_wallet.py

```diff
@@ -711,16 +711,31 @@ async def get_latest_singleton(self, launcher_id: bytes32) -> Optional[Singleton
         )
         return singleton
 
-    async def get_history(self, launcher_id: bytes32) -> List[SingletonRecord]:
+    async def get_history(
+        self,
+        launcher_id: bytes32,
+        min_generation: Optional[uint32] = None,
+        max_generation: Optional[uint32] = None,
+        num_results: Optional[uint32] = None,
+    ) -> List[SingletonRecord]:
         history: List[SingletonRecord] = await self.wallet_state_manager.dl_store.get_all_singletons_for_launcher(
-            launcher_id
+            launcher_id,
```

Comment:
maybe change to dl_wallet_store to avoid confusion with the data_store ?

Reply:
I have a local branch for this and other changes that I'll PR when general changes will break less other PRs.